app/perfunwind.h

```diff
@@ -94,11 +95,12 @@ class PerfUnwind : public QObject
     };
 
     struct UnwindInfo {
-        UnwindInfo() : frames(0), unwind(nullptr), sample(nullptr), maxFrames(64),
+        UnwindInfo() : frames(0), disasmFrames(0), unwind(nullptr), sample(nullptr), maxFrames(64),
             firstGuessedFrame(-1), isInterworking(false) {}
 
         QHash<qint32, QHash<quint64, Dwarf_Word>> stackValues;
         QVector<qint32> frames;
```

Comment:
what is the difference between frames and disasmFrames - this needs work.

Reply:
There is the difference for LBR unwinding method. 
Some required frames are not looked up for LBR.
So I have entered disasmFrames to accumulate frames for LBR to compute then Disassembly events costs within function/method.

Reply:
I'm sorry but I don't understand this - can you give a more concrete example maybe that shows the actual problem you are trying to solve here?

Reply:
The problem is available on any perf.data recorded with LBR unwinding method. 
For LBR only branchStack is traversed if it is presented. 
But it is not enough to compute events costs for assembler instructions.
I accumulate such 'missed' frames in disasmFrames for LBR.
There is no any problem for other unwing methods such as fp, dwarf.

Reply:
sorry, but I still don't follow this - can you give a concrete example please? Ideally by writing a proper unit test for this case - you can try to run perf record with LBR on one of the static binaries in the test directory. Then expand the test output to also include the info for the disassembler view. All of that could be done in the previous patches. This one would then show in the changed test expectations what frames we are missing.

I mean - once again - from my understanding: if LBR is enabled for recording, one only uses that for the user space frames. The FP frames may still be there when we we switch from user space to the kernel (afaik?). But that's independent of the disassembler, and I thought we handle this correctly already. 

Put differently: When/how would the disassembler require more/different frames, that differs from the rest of the tools and views? Why would the frames for the disassembler be different from, say, the flame graph view?

Reply:
I put example to the bottom of conversation. If you record perf.data with --call-graph lbr you will get traverse only branchStack because there is 'break' from callchain traverse at all if it hasBranchStack. I have just moved that 'break' at the end of first iteration of callchain traverse.